cerberus-api-tests/src/main/groovy/com/nike/cerberus/api/CerberusApiActions.groovy

```diff
@@ -93,13 +104,19 @@ class CerberusApiActions {
     }
 
     static def getDecryptedPayload(String iamPrincipalArn, String region, String base64EncodedKmsEncryptedAuthPayload, boolean assumeRole = true) {
-        AWSKMSClient kmsClient
-        if (assumeRole) {
-            kmsClient = new AWSKMSClient(new STSProfileCredentialsServiceProvider(
-                    new RoleInfo().withRoleArn(iamPrincipalArn)
-                            .withRoleSessionName(UUID.randomUUID().toString()))).withRegion(Regions.fromName(region))
-        } else {
-            kmsClient = new AWSKMSClient().withRegion(Regions.fromName(region))
+        AWSKMSClient kmsClient = kmsClientCache.getIfPresent(new Tuple2(iamPrincipalArn, region))
```

Comment:
Yes, I verified the implementation of Tuple2.  Both the equals() and hashcode() methods looked right to me for what I'm doing here.  

- The equals() method in Tuple2 will call String#equals() on both elements.
- The hashcode() method in Tuple2 will use String#hashcode() on both elements.